exporter/splunkhecexporter/integration_test.go

```diff
@@ -74,17 +73,6 @@ func teardown(cfg SplunkContainerConfig) {
 	fmt.Printf("Command output:\n%s\n", output)
 }
 
-func TestMain(m *testing.M) {
```

Comment:
Since this `TestMain` was declared in an `integration` test, it isn't noticed until running with the integration tag set. I decided to keep the uniform declaration of `TestMain` in a new `package_test.go` file for `goleak` checks. The down side here is test execution will take longer and will be doing duplicate work by setting up and tearing down with each test.

Alternatives here would be keep `TestMain` in the integration test and do goleak checks there. This works but goleak checks would only run with integration tests, and it would be less obvious it's running.

I'm fine with either alternative, just let me if anyone has a preference.

Reply:
Are you saying put them [all into one test function](https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks), then use `t.Run` to call each? (I want to make sure I understand exactly what you mean before making changes.)

Reply:
I think that's what he means.

Reply:
I've refactored as requested. 👍

Reply:
Note: All test functionality should be the same. I moved some things around, fixed some asserts to be more descriptive, and fixed some typos.